sympy/sets/fancysets.py

```diff
@@ -667,6 +668,12 @@ def __new__(cls, sets, polar=False):
             obj = ImageSet.__new__(cls, Lambda((r, theta),
                                    r*(cos(theta) + I*sin(theta))),
                                    sets)
+            obj._variables = (r, theta)
+            obj._expr = r*(cos(theta) + I*sin(theta))
+
+        obj._sets = sets
+        obj._polar = polar
+        obj._args = (sets, polar)
```

Comment:
There are some warnings about dealing with ``_args`` in basic.py. (hash often depends on it.) It would be possible to define a separate ``args`` property that would return ``(self._sets, self._polar)``.